compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -491,6 +498,29 @@ impl<'a> StringReader<'a> {
         FatalError.raise()
     }
 
+    fn report_reserved_prefix(&self, start: BytePos) {
+        // See RFC 3101.
+        if self.sess.edition < Edition::Edition2021 {
+            return;
+        }
+
+        let mut err = self.sess.span_diagnostic.struct_span_err(
+            self.mk_sp(start, self.pos),
+            &format!("prefix `{}` is unknown", self.str_from_to(start, self.pos)),
+        );
+        err.span_label(self.mk_sp(start, self.pos), "unknown prefix");
+        err.span_label(
+            self.mk_sp(self.pos, self.pos),
+            &format!(
+                "help: consider inserting a whitespace before this `{}`",
+                self.str_from_to(self.pos, self.pos + BytePos(1)),
+            ),
+        );
```

Comment:
You can use `span_suggestion_vebose` to show the suggestion directly.
```suggestion
        err.span_suggestion_verbose(
            self.mk_sp(self.pos, self.pos),
            "consider inserting whitespace here",
            " ".into(),
            Applicability::MachineApplicable,
        );
```

Output:

![image](https://user-images.githubusercontent.com/783247/121891542-17a5ac80-cd0b-11eb-8fa7-c9f3245a9e02.png)